src/main/resources/templates/pet/edit.html

```diff
@@ -1,128 +1,168 @@
-                                        th:value="${status}" th:text="${status.value}" th:selected="(${status} == *{status})"/>
-                                </select>
-                            <br/>
-                            </div>
-                            <label for="type"><h4 th:text="#{pet.type}"/></label>
-                            <select id="typeSelector" name="type" th:field="*{type}" class="form-control"
+                                <br />
+                                <label for="image">
+                                    <h4 th:text="#{pet.image}" />
+                                </label>
+                                <input type="file" name="image" />
+                                <br />
+                                <label for="birthDate">
+                                    <h4 th:text="#{pet.birthDate}" />
+                                </label>
+                                <input type="datetime-local" id="birthDate" name="birthDate" th:field="*{birthDate}">
+                                <label th:if="${#fields.hasErrors('birthDate')}" th:errors="*{birthDate}"></label>
+                                <br />
+                                <label for="dewormed">
+                                    <h4 th:text="#{pet.dewormed}" />
+                                </label>
+                                <input type="checkbox" name='dewormed' th:field="*{dewormed}" class="form-control"
+                                    placeholder="dewormed" id='dewormed' />
+                                <label th:if="${#fields.hasErrors('dewormed')}" th:errors="*{dewormed}"></label>
+                                <br />
+                                <label for="sterilized">
+                                    <h4 th:text="#{pet.sterilized}" />
+                                </label>
+                                <input type="checkbox" name='sterilized' th:field="*{sterilized}" class="form-control"
+                                    placeholder="sterilized" id='sterilized' />
+                                <label th:if="${#fields.hasErrors('sterilized')}" th:errors="*{sterilized}"></label>
+                                <br />
+                                <h4>Vaccination Records</h4>
+                                <table class="table">
+                                    <thead>
+                                        <tr>
+                                            <th>Vaccine</th>
+                                            <th>Status</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        <!-- Loop through the pet's vaccination records -->
+                                        <tr th:each="vaccine : ${petCommand.vaccinationRecords}">
```

Comment:
Seems like `vaccinationRecords` do not exist in `PetCommand`, you might need to add and populate.
Also validate if the list is empty.